src/editor/ui_menus/tool_place_immovable_options_menu.cc

```diff
@@ -33,17 +33,24 @@ namespace {
 UI::Checkbox* create_immovable_checkbox(UI::Panel* parent,
                                         LuaInterface* lua,
                                         const Widelands::ImmovableDescr& immovable_descr) {
-	const Image* pic = immovable_descr.representative_image();
+      const Image* pic = immovable_descr.representative_image();
 
-	std::string tooltip = immovable_descr.descname();
+   std::string tooltip;
 
 	// Get information about preferred terrains
 	if (immovable_descr.has_terrain_affinity()) {
 		std::unique_ptr<LuaTable> table(lua->run_script("scripting/editor/tree_tooltip.lua"));
 		std::unique_ptr<LuaCoroutine> cr(table->get_coroutine("func"));
 		cr->push_arg(immovable_descr.name());
 		cr->resume();
-		tooltip.append(cr->pop_table()->get_string("text"));
+		tooltip = format(_("%1$s %2$s"), immovable_descr.descname(), cr->pop_table()->get_string("text"));
+	} else {
+		std::string size = Widelands::BaseImmovable::size_to_string(immovable_descr.get_size());
+		if (size == "none") {
+			tooltip = format(_("%1$s (%2$s)"), immovable_descr.descname(), _("removable by e.g. placing roads"));
+		} else {
+			tooltip = format(_("%1$s (%2$s %3$s %4$s)"), immovable_descr.descname(), _("needs a"), size, _("building plot"));
```

Comment:
Better, but one abstraction too much now :) 
```c++
			tooltip = format(_("%1$s (needs a %2$s building plot)"), immovable_descr.descname(), size);
```

However I just checked and it turns out that `size_to_string` is not localized. So this would result in e.g. "Braucht einen small Bauplatz". Instead of calling `size_to_string`, better use a `switch` statement over `immovable_descr.get_size()` to create a different hardcoded string for each of the four size constants `Widelands::BaseImmovable::NONE|SMALL|MEDIUM|BIG`.

Reply:
> ... better use a switch statement over immovable_descr.get_size() to create a different hardcoded string for each of the four size constants

Not sure if i understand correctly: Do you mean overloading this method?

Reply:
No:
```c++
switch (immovable_descr.get_size()) {
case Widelands::BaseImmovable::NONE:
	tooltip = format(_("%1$s (removable by e.g. placing roads)"), immovable_descr.descname());
	break;
case Widelands::BaseImmovable::SMALL:
	tooltip = format(_("%1$s (needs a small building plot)"), immovable_descr.descname());
	break;
... // similar for MEDIUM and BIG
```